knodle/trainer/snorkel/utils.py

```diff
@@ -13,4 +18,52 @@ def z_t_matrix_to_snorkel_matrix(rule_matches_z: np.ndarray, mapping_rules_label
         non_zero_idx = np.where(rule_matches_z[i] != 0)[0]
         snorkel_matrix[i, non_zero_idx] = z_to_t[non_zero_idx]
 
-    return snorkel_matrix
\ No newline at end of file
+    return snorkel_matrix
+
+def prepare_empty_rule_matches(rule_matches_z, filter_non_labelled, model_input_x
+) -> Tuple[np.ndarray, np.ndarray, TensorDataset]:
+    """
+    Remove empty rows in rule matches for LabelModel. If filtering is configured, the corresponding entries
+    in the model input are filtered out as well.
+    Args:
+        rule_matches_z:
+        filter_non_labelled: if rows with no rule matches should be filtered.
+        model_input_x:
+    Returns:
+        boolean mask indicating non-empty rows,
+        filtered rule matches,
+        and eventually filtered model input
+    """
+    # find empty rows of the rule matches
+    non_zero_mask = rule_matches_z.sum(axis=1) != 0
+    non_zero_indices = np.where(non_zero_mask)[0]
+
+    # exclude empty rows from LabelModel input
+    rule_matches_z = rule_matches_z[non_zero_indices]
```

Comment:
the empty rows are filtered out from `rule_matches_z` in all cases? That means: 
- the `rule_matches_z` and `model_input_x` outputs of this function may have not the same dimension ?
- `L_train` length may be smaller that number of samples in `model_input_x` ?